galsim/fitswcs.py

```diff
@@ -224,13 +224,19 @@ def _radec(self, x, y, color=None):
         return ra, dec
 
     def _xy(self, ra, dec, color=None):
-        import astropy
         factor = radians / degrees
-        rd = np.atleast_2d([ra, dec]) * factor
+
+        r1 = np.atleast_1d(ra) * factor
+        d1 = np.atleast_1d(dec) * factor
+
         with warnings.catch_warnings():
             warnings.simplefilter("ignore")
-            xy = self.wcs.all_world2pix(rd, 1, ra_dec_order=True)[0]
-        x, y = xy
+            x, y = self.wcs.all_world2pix(r1, d1, 1, ra_dec_order=True)
+        try:
+            len(ra)
+        except TypeError:
```

Comment:
> Also, does the code allow ra, dec that can be broadcasted together? In that case this might fail in weird ways.

Not sure what you mean by this.  Can you give me an example of what you have in mind?

Reply:
Right now I think this might give weird results, but IDK if the API allows these inputs

```
self._radec(10, np.array([34, 56, 78]), units='degrees')
```

Reply:
I see.  Currently, this sometimes goes through and sometimes gives errors.  (I just wrote a unit test to check.)  And some of the time it goes through it gives the right answer, but sometimes not.  
Do you think this is something that a user is likely to try?  If so, should I work to make it always return the right thing, or check and raise an exception?

Reply:
I think this depends on how seriously we take the numpy API. Generally, the numpy API supports any combination of operations that broadcast correctly. 

I'd say we either support broadcasting, or explicitly check that either both arrays are the same size or both are scalars. A silent failure here seems to be pretty catastrophic in that it sometimes gives the wrong answers.

In terms of whether or not this is common, I honestly have no idea. I think some extra error checking is the best way to go here mostly because that seems like the least amount of effort and I am happy to make sure the inputs conform for my own code.

Reply:
Hm..   This seems like a bit more handholding than I generally want to do for the user.  There's a balance between checking for likely mistakes and giving helpful errors vs slowing down the code by checking for every possible thing someone could do wrong.  (if statements in python aren't free.)
My compromise now is to include a couple of `assert` statements that the inputs are the same shape at this point when returning numpy arrays.  These tend to be faster than if statements in python and still serve the purpose of telling the user than their code is wrong without bending over backwards to give a helpful error message.  (They can also be turned off with -O if the run is really performance critical, but nobody usually does this.)